### DISCUSSION TITLE
ComboBox / autocomplete with filtering dropdown options.

### DISCUSSION THREAD

**Discussion by pbosin** [microsoft/fluentui#19153] [Q&A]
Looking for a way to have a dropdown with the button text input like in a ComboBox, which would filter the dropdown options as you type. 
The documentation at https://developer.microsoft.com/en-us/fluentui#/controls/web/combobox shows ComboBox with toggleable autoComplete and allowFreeform. But it does not behave in the expected way. When you start typing in the combo box, the options in the dropdown do not change and only the first matching option is highlighted.
Is there a way to configure it for show only the matching options?

**Comment by sylvain-hamel** [comment]
I have the same use case. My `ComboBox` shows around 200 options. Finding the right option is difficult even after typing the first few characters. I suppose `autoComplete` is not exactly what I'm looking for, more like `autoFilter`. I can't use a `Dropdown` because I need the user to be able to enter text if they don't find what they want in the list.

**Comment by Pankaj-capsitech** [comment]
I have the same use case. Please suggest any way for this

**Comment by 15MariamS** [comment]
The docs have a good example of [filtering](https://react.fluentui.dev/?path=/docs/components-combobox--default#filtering). I recommend changing `indexOf` to `includes` in the filter as detailed below.

[Codesandbox](https://codesandbox.io/s/hjmw29?file=/example.tsx):
```
const onChange: ComboboxProps["onChange"] = (event) => {
    const value = event.target.value.trim();
    const matches = options.filter(
      (option) => option.toLowerCase().indexOf(value.toLowerCase()) === 0
    );
    setMatchingOptions(matches);
  };
```

If you want to match on any term rather than having to type the words in order, you can change it [like this](https://codesandbox.io/s/solitary-pine-62wjd5?file=/example.tsx:1198-1450):

```
  const onChange: ComboboxProps["onChange"] = (event) => {
    const value = event.target.value.trim().toLowerCase();
    const matches = options.filter((option) =>
      option.toLowerCase().includes(value)
    );
    setMatchingOptions(matches);
  };
```
<img width="283" alt="Screenshot 2023-10-28 at 1 57 24 PM" src="https://github.com/microsoft/fluentui/assets/81782111/ff2e35ea-3e39-4e8a-85c2-227fb140e2d9">
